src/test/java/de/tum/in/www1/artemis/service/notifications/ConversationNotificationServiceTest.java

```diff
@@ -0,0 +1,100 @@
+package de.tum.in.www1.artemis.service.notifications;
+
+import static de.tum.in.www1.artemis.domain.notification.NotificationTitleTypeConstants.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.metis.ConversationParticipant;
+import de.tum.in.www1.artemis.domain.metis.Post;
+import de.tum.in.www1.artemis.domain.metis.conversation.OneToOneChat;
+import de.tum.in.www1.artemis.domain.notification.ConversationNotification;
+import de.tum.in.www1.artemis.domain.notification.Notification;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationMessageRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.repository.metis.conversation.ConversationNotificationRepository;
+import de.tum.in.www1.artemis.repository.metis.conversation.ConversationRepository;
+
+class ConversationNotificationServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    private static final String TEST_PREFIX = "conversationnotificationservice";
+
+    @Autowired
+    private ConversationNotificationRepository conversationNotificationRepository;
+
+    @Autowired
+    private ConversationNotificationService conversationNotificationService;
+
+    @Autowired
+    private ConversationMessageRepository conversationMessageRepository;
+
+    @Autowired
+    private ConversationRepository conversationRepository;
+
+    @Autowired
+    private ConversationParticipantRepository conversationParticipantRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    private OneToOneChat oneToOneChat;
+
+    private User user1;
+
+    @BeforeEach
+    void setUp() {
+        this.database.addUsers(TEST_PREFIX, 5, 1, 0, 1);
```

Comment:
do we really need so many users? this will make the test execution long

Reply:
We actually don't need, i will reduce to 2 users